src/Propel/Generator/Model/Table.php

```diff
@@ -784,6 +784,25 @@ public function setupReferrers($throwErrors = false)
                 }
             }
 
+            // check for incomplete foreign key references when foreign table
+            // has a composite primary key
+            if ($foreignTable->hasCompositePrimaryKey()) {
+                // get composite foreign key's keys
+                $foreignPrimaryKeys = $foreignTable->getPrimaryKey();
+                // check all keys are referenced in foreign key
+                foreach ($foreignPrimaryKeys as $foreignPrimaryKey) {
+                    if (!$foreignPrimaryKey->hasReferrer($foreignKey) && $throwErrors) {
```

Comment:
What if my foreign key refers to an unique key of $foreignTable instead of its primary key? This should be supported as well.